examples/cloudrun/variables.tf

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  type = string
+}
+
+variable "region" {
+  description = "Location for load balancer and Cloud Run resources"
+  default     = "us-central1"
+}
+
+variable "ssl" {
+  description = "Run load balancer on HTTPS and provision managed certificate with provided `domain`."
+  type        = bool
+  default     = true
```

Comment:
Should we change the default to false to match the option 1?

Reply:
I prefer we keep the default=true, but have ssl=false to illustrate "insecure mode" if that's OK. We can drop `ssl=true` in the example.

Reply:
If I set ssl = true but don't specify any domains, will it work? Generally speaking we strive to avoid situations where an example config has *implicit* requirements - I should be able to run `terraform apply` with only the required variables (`project`) and get a working output.

Reply:
It won't work. I default domain to `""` so it works with ssl=false.
I think the best solution is users stick to the instructions. I'm assuming examples/cloudrun is not designed to be an importable module but more like a standalone example.

Reply:
In that case, we should either change the default (my preference) or remove the default for domain. I recognize that we want to encourage SSL but I'd rather have an example users can start with.

Yes, it's ideal if users stick to instructions but there's also a common expectation that if you `terraform apply` an example and fill in all _required_ variables you should get a working config. Implicitly required variables are an anti-pattern.

I recognize it's not meant to be an importable module, this is mostly for the experience of users who just copy+paste or checkout the example.

Reply:
I'm guessing >90% of the intended audience will end up using `-ssl=true`, so I think we shouldn't change default to `ssl=false`.

Is `ssl=true` default and "required" `domain` acceptable? This way users probably have to do something like `-var=domain=ignored` or `-var=domain=null` to fulfill the requirement?

Reply:
Yes that's acceptable, though keep in mind the example doesn't necessarily need to match people's eventual configuration.

Reply:
I think I've now addressed this.

Reply:
Thanks, yeah this should work.